### ISSUE TITLE
New Settings - Account Templates - cannot create fresh Account settings

### ISSUE THREAD

**Issue by RealRaven2000** [RealRaven2000/SmartTemplates#300]
When creating / saving settings for a new account these are not saved. Instead we see the following errors in error console.

```log
WebExtensions: Unknown legacy preference <extensions.smartTemplate4.id8.newmsg>, forgot to declare a default?. [implementation.js:169](jar:file:///E:/Mail/dd9jwla2n.axel/extensions/smarttemplate4@thunderbird.extension.xpi!/chrome/content/api/LegacyPrefs/implementation.js)
WebExtensions: Unknown legacy preference <extensions.smartTemplate4.id8.rspmsg>, forgot to declare a default?. [implementation.js:169](jar:file:///E:/Mail/dd9jwla2n.axel/extensions/smarttemplate4@thunderbird.extension.xpi!/chrome/content/api/LegacyPrefs/implementation.js)
WebExtensions: Unknown legacy preference <extensions.smartTemplate4.id8.rspheader>, forgot to declare a default?. [implementation.js:169](jar:file:///E:/Mail/dd9jwla2n.axel/extensions/smarttemplate4@thunderbird.extension.xpi!/chrome/content/api/LegacyPrefs/implementation.js)
WebExtensions: Unknown legacy preference <extensions.smartTemplate4.id8.fwdmsg>, forgot to declare a default?.
```

related: issue #259

**Comment by RealRaven2000** [comment]
Here is a version that is able to create the missing settings:

[smartTemplate-fx-4.6.2pre3.zip](https://github.com/user-attachments/files/16283549/smartTemplate-fx-4.6.2pre3.zip)

for this I made patch in the LegacyPrefs API.

---

to install the version above, drag the archive (zip file) into Thunderbird Add-ons manager - do not unpack contents as then it won't install.

**Comment by RealRaven2000** [comment]
I had a meeting with @jobisoft yesterday and asked him to look at my patch for the LegacyPrefs experimental API, I believe he will patch in the changes todays so I can get an official version for integration into SmartTemplates 4.6.2 so that I can do a maintenance update.

**Comment by RealRaven2000** [comment]
Fixed in 4.7 - published 05/08/2024
